Microsoft.SystemForCrossDomainIdentityManagement/Service/BulkUpdateOperationState.cs

```diff
@@ -0,0 +1,230 @@
+
+        private void Fault(HttpStatusCode statusCode, ErrorType? errorType = null)
+        {
+            ErrorResponse error =
+                        new ErrorResponse()
+                        {
+                            Status = statusCode
+                        };
+            if (errorType.HasValue)
+            {
+                error.ErrorType = errorType.Value;
+            }
+            BulkResponseOperation response =
+                new BulkResponseOperation(this.Operation.Identifier)
+                {
+                    Response = error
+                };
+            this.Complete(response);
+        }
+
+        public override bool TryPrepareRequest(out IRequest<IPatch> request)
+        {
+            request = null;
+
+            PatchRequest2 patchRequest;
+            switch (this.Operation.Data)
+            {
+                case PatchRequest2 patchrequest2:
+                    patchRequest = patchrequest2;
+                    break;
+                default:
+                    dynamic operationDataJson = JsonConvert.DeserializeObject(Operation.Data.ToString());
+                    var patchOperations = operationDataJson.Operations.ToObject<List<PatchOperation2Combined>>();
+                    patchRequest = new PatchRequest2(patchOperations);
+                    break;
+            }
+            IPatch patch =
+                new Patch()
+                {
+                    PatchRequest = patchRequest
+                };
+            IRequest<IPatch> requestBuffer =
+                new UpdateRequest(
+                    this.BulkRequest.Request,
+                    patch,
+                    this.BulkRequest.CorrelationIdentifier,
+                    this.BulkRequest.Extensions);
+
+            Uri resourceIdentifier;
+            if (this.Parent != null)
+            {
+                if (null == this.Parent.Response || null == this.Parent.Response.Location)
+                {
+                    this.Fault(HttpStatusCode.NotFound, ErrorType.noTarget);
+                    return false;
+                }
+
+                resourceIdentifier = this.Parent.Response.Location;
+            }
+            else
+            {
+                if (null == this.BulkRequest || null == this.BulkRequest.BaseResourceIdentifier)
+                {
+                    throw new InvalidOperationException(SystemForCrossDomainIdentityManagementServiceResources.ExceptionInvalidState);
+                }
+
+                if (null == this.Operation.Path)
+                {
+                    this.Fault(HttpStatusCode.BadRequest);
+                    return false;
+                }
+
+                resourceIdentifier = new Uri(this.BulkRequest.BaseResourceIdentifier, this.Operation.Path);
+            }
+
+            if
+            (
+                    !UniformResourceIdentifier.TryParse(resourceIdentifier, this.BulkRequest.Extensions, out IUniformResourceIdentifier parsedIdentifier)
```

Comment:
This line may extend over the 135 character right margin that is recommended as the limit for the sake of screen magnifiers used by the sight-impaired.